src/libcore/num/mod.rs

```diff
@@ -2367,9 +2373,7 @@ macro_rules! uint_impl {
         /// ```
         #[stable(feature = "rust1", since = "1.0.0")]
         pub fn checked_next_power_of_two(self) -> Option<Self> {
-            let bits = size_of::<Self>() * 8;
-            let one: Self = 1;
-            let npot = one << ((bits - self.wrapping_sub(one).leading_zeros() as usize) % bits);
+            let npot = self.one_less_than_next_power_of_two().wrapping_add(1);
```

Comment:
Instead of the manual some/none, can it just be this?
```rust
pub fn checked_next_power_of_two(self) -> Option<Self> {
    self.one_less_than_next_power_of_two()
        .checked_add(1)
}
```